cli/command/container/utils.go

```diff
@@ -53,27 +75,20 @@ func waitExitOrRemoved(ctx context.Context, dockerCli *command.DockerCli, contai
 		return false
 	}
 
-	// Get events via Events API
-	f := filters.NewArgs()
-	f.Add("type", "container")
-	f.Add("container", containerID)
-	options := types.EventsOptions{
-		Filters: f,
-	}
-
-	eventCtx, cancel := context.WithCancel(ctx)
-	eventq, errq := dockerCli.Client().Events(eventCtx, options)
-
 	go func() {
 		defer cancel()
 
 		for {
 			select {
+			case <-eventCtx.Done():
+				if removeErr != nil {
+					logrus.Errorf("error removing container: %v", removeErr)
+					return
+				}
```

Comment:
Also, I think if `ContainerRemove` failed and `removeError != nil`, client will still hang. Should we send exitcode to statusChan to make client exit? I mean 
```
				if removeErr != nil {
					logrus.Errorf("error removing container: %v", removeErr)
+					statusChan <- exitCode
					return
				}
```